lms/djangoapps/course_api/blocks/permissions.py

```diff
@@ -24,11 +29,25 @@ def can_access_others_blocks(requesting_user, course_key):
     return has_access(requesting_user, CourseStaffRole.ROLE, course_key)
 
 
-def can_access_self_blocks(requesting_user, course_key):
+def can_access_self_blocks(requesting_user: User, course_key: CourseKey) -> AccessResponse:
     """
     Returns whether the requesting_user can access own blocks.
     """
-    return (
+    user_is_enrolled_or_staff = (  # pylint: disable=consider-using-ternary
         (requesting_user.id and CourseEnrollment.is_enrolled(requesting_user, course_key)) or
         has_access(requesting_user, CourseStaffRole.ROLE, course_key)
     )
+    if user_is_enrolled_or_staff:
+        return ACCESS_GRANTED
+    try:
+        return is_course_public(course_key)
+    except ValueError:
+        return ACCESS_DENIED
+
+
+def is_course_public(course_key: CourseKey) -> AccessResponse:
```

Comment:
Moved this here, because importing get_course in access_utils leads to a whole range of circular imports. If it really should be in access_utils I can make an import in this function to avoid the circular imports.